### PR TITLE
Added NSX-T Transport Nodes

### PR DESCRIPTION
**Pull request by richardtief**
Transport Zones and Transport Nodes have been added to enhance NSX-T monitoring. These can be used to capture metrics, properties and alerts for these objects. 

plus additional bugfixes

https://docs.vmware.com/en/VMware-NSX-T-Data-Center/3.1/installation/GUID-CDD3A8D5-5D90-41D8-B532-28AE0E5F20C9.html

### CODE REVIEW

File: InventoryBuilder.py [hunk 1]
```diff
@@ -283,11 +288,15 @@ def create_nsxt_objects(self, vrops, target: str, token: str):
...
-            Vrops.get_nsxt_mgmt_cluster(vrops, target,token,[a.uuid for a in nsxt_adapter])
+            Vrops.get_nsxt_mgmt_cluster(vrops, target, token, [a.uuid for a in nsxt_adapter])
         nsxt_mgmt_nodes, self.response_codes[target]["nsxt_mgmt_nodes"] = \
             Vrops.get_nsxt_mgmt_nodes(vrops, target, token, [c.uuid for c in nsxt_mgmt_cluster])
         nsxt_mgmt_service, self.response_codes[target]["nsxt_mgmt_services"] = \
             Vrops.get_nsxt_mgmt_service(vrops, target, token, [n.uuid for n in nsxt_mgmt_nodes])
+        nsxt_transport_zones, self.response_codes[target]["nsxt_transport_zones"] = \
+            Vrops.get_nsxt_transport_zone(vrops, target, token, [c.uuid for c in nsxt_mgmt_cluster])
+        nsxt_transport_nodes, self.response_codes[target]["nsxt_transport_nodes"] = \
+            Vrops.get_nsxt_transport_node(vrops, target, token, [t.uuid for t in nsxt_transport_zones])
```
Comment (viennaa) [inline InventoryBuilder.py:299]:
```suggestion
            Vrops.get_nsxt_transport_node(vrops, target, token, [z.uuid for z in nsxt_transport_zones])
```
I would feel `z` more natural here, looking at the previous line pattern.

File: InventoryBuilder.py [hunk 2]
```diff
@@ -50,43 +50,43 @@ def vrops_list():
 
         @app.route('/<target>/vcenters/<int:iteration>', methods=['GET'])
         def vcenters(target, iteration):
-            return self.iterated_inventory[str(iteration)]['vcenters'].get(target, {})
+            return self.iterated_inventory.get(str(iteration)).get('vcenters').get(target, {})
```
Comment (viennaa) [inline InventoryBuilder.py:53]: Did you face any KeyErrors here?

**Review by viennaa — COMMENTED** [review]
Thanks for providing this PR. Just two little things to clarify before we can proceed.

File: InventoryBuilder.py (hunk 2 above)
Comment (richardtief) [reply InventoryBuilder.py:53]: Yes, especially when the inventory pod restarts.

**Review by viennaa — APPROVED** [review]
More solid, more better. Let's go!

**Pull request merged**